plugins/nf-nomad/src/main/nextflow/nomad/executor/NomadTaskHandler.groovy

```diff
@@ -89,15 +116,14 @@ class NomadTaskHandler extends TaskHandler implements FusionAwareTask {
 
     @Override
     boolean checkIfCompleted() {
-        if( !jobName ) throw new IllegalStateException("Missing Nomad Job name -- cannot check if running")
+        if( !jobName ) throw new IllegalStateException("[NOMAD] Missing Nomad Job name -- cannot check if running")
+        def isFinished = false
 
         def state = taskState0()
 
-        final isFinished = state && (state.finishedAt != null || state.state == "unknow")
-
-        log.debug "[NOMAD] checkIfCompleted task.name=$task.name; state=${state?.state} completed=$isFinished"
+        log.debug "[NOMAD] checkIfCompleted task=$task.name; state=${state?.state}"
 
-        if (isFinished) {
+        if( state && ( ["dead","complete"].contains(state.state))){
```

Comment:
@matthdsm  , the problem in #97  is due to incomplete mapping of Nomad job/task statuses. I still don't quite understand it well enough, but I trust that marking the `unknown` status as `TaskStatus.RUNNING` should solve the problem you're facing. 



CC @jagedn